ACCS-365: Payment Services: Stored Cards at Checkout (#821)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/payment-services/functions.mdx

*/}

import TableWrapper from '@components/TableWrapper.astro';

> The Payment Services drop-in exposes GraphQL helpers for vaulted (saved) card checkout. Additional functions may appear in the generated section below when sourced from the drop-in repository.

Reviewer: ```suggestion
The Payment Services drop-in provides GraphQL helpers for vaulted (saved) card checkout. Additional functions may appear in the generated section below when sourced from the drop-in repository.
```

<TableWrapper nowrap={[0]}>

| Function | Description |
| --- | --- |
| [`getCustomerPaymentTokens`](#getcustomerpaymenttokens) | Loads vaulted Payment Services card tokens for the signed-in customer via the `customerPaymentTokens` query. |
| [`createPaymentOrderForVault`](#createpaymentorderforvault) | Creates a PayPal / MP payment order for vault checkout using `createPaymentOrder`. |
| [`syncVaultToCart`](#syncvaulttocart) | Runs vault `createPaymentOrder` then `setPaymentMethod` using your checkout `fetchGraphQl` and cart APIs. |
| [`normalizedVaultTokenToStoredCardProps`](#normalizedvaulttokentostoredcardprops) | Maps a `NormalizedVaultToken` to props for the [StoredCards](/dropins/payment-services/containers/stored-cards/) container. |

</TableWrapper>

## getCustomerPaymentTokens

> The `getCustomerPaymentTokens` function loads customer payment tokens using this package’s GraphQL client (GET). The initializer must set the endpoint and headers (see [Initialization](/dropins/payment-services/initialization/)). Results are filtered to Payment Services vaulted card tokens and returned as **`NormalizedVaultToken[]`**.

Reviewer: ```suggestion
The `getCustomerPaymentTokens` function retrieves customer payment tokens using this package’s GraphQL client (GET). The initializer must configure the endpoint and headers (see [Initialization](/dropins/payment-services/initialization/)). Results are filtered to Payment Services vaulted card tokens and returned as **`NormalizedVaultToken[]`**.
```

```ts
export async function getCustomerPaymentTokens(): Promise<NormalizedVaultToken[]>
```

### Returns

Returns **`NormalizedVaultToken[]`**. See the [StoredCards container](/dropins/payment-services/containers/stored-cards/) and [`normalizedVaultTokenToStoredCardProps`](#normalizedvaulttokentostoredcardprops) (including **Related types**).

## createPaymentOrderForVault

> The `createPaymentOrderForVault` function calls the vault **`createPaymentOrder`** mutation and returns PayPal and MP order identifiers for use in payment method payload construction.

Reviewer: ```suggestion
The `createPaymentOrderForVault` function calls the vault `createPaymentOrder` mutation and returns the PayPal and MP order IDs used to build the payment method payload.
```

```ts
export async function createPaymentOrderForVault(
  cartId: string
): Promise<VaultPaymentOrderIds>
```

<TableWrapper nowrap={[0, 1]}>

| Parameter | Type | Req? | Description |
| --- | --- | --- | --- |
| `cartId` | `string` | Yes | Active cart id for the checkout session. |

</TableWrapper>

### Returns

Returns **`{ paypalOrderId, paymentsOrderId }`**. Throws if the mutation returns no order ids.

## syncVaultToCart

> The `syncVaultToCart` function orchestrates vault checkout for storefronts that supply checkout (or mesh) **`fetchGraphQl`** and cart **`setPaymentMethod`**. It runs the same **`createPaymentOrder`** flow as **`createPaymentOrderForVault`**, builds vault additional data with the token’s **`public_hash`**, then calls **`setPaymentMethod`** with **`PaymentMethodCode.VAULT`**.

Reviewer: ```suggestion
The `syncVaultToCart` function manages the vault checkout flow for storefronts that provide checkout (or mesh) `fetchGraphQl` and cart `setPaymentMethod`. It follows the same `createPaymentOrder` flow as `createPaymentOrderForVault`, builds the vault-specific additional data using the token’s `public_hash`, and then calls `setPaymentMethod` with `PaymentMethodCode.VAULT`.
```
